### PR TITLE
Ndk packets refactor

### PR DESCRIPTION
**Pull request by Div-Inworld**
Complete Working NDK with updated DLL and factory to create packets

### CODE REVIEW

**Comment by realill** [comment]
Are we merging it into main?

File: Assets/Inworld.AI/Scripts/Runtime/Entities/NDK/Packets_Utility/ControlEvent.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        //     };
+        // }
+
+        // public ControlEvent(PacketId id, DateTime timestamp, GrpcControlEvent.Types.Action action, Routing routing): this()
+        // {
+        //     PacketId = id;
+        //     Timestamp = timestamp;
+        //     Action = action;
+        //     Routing = routing;
+        // }
```
Comment (realill) [inline Assets/Inworld.AI/Scripts/Runtime/Entities/NDK/Packets_Utility/ControlEvent.cs:57]: Should keep any commented out code, if do should come with TODO on what to do in future.

File: Assets/Inworld.AI/Scripts/Runtime/Entities/NDK/InworldPacketGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            if (_instance == null)
+            {
+                _instance = new InworldPacketGenerator();
+            }
+
+            return _instance;
+        }
+    }
+
+    public ProtoPacket ToProtobufPacket(InworldPacket packet)
```
Comment (realill) [inline Assets/Inworld.AI/Scripts/Runtime/Entities/NDK/InworldPacketGenerator.cs:61]: You should add test for this class converting to->from and then comparing results.

**Comment by Div-Inworld** [comment]
> Are we merging it into main?

maybe not yet. Was mostly just for comparison so Yan could see the changes clearly

**Pull request closed without merging**